event/src/anchor.rs

```diff
@@ -0,0 +1,86 @@
+use cid::Cid;
+use indexmap::IndexMap;
+use serde::{Deserialize, Serialize};
+
+use crate::unvalidated::{Proof, RawTimeEvent};
+
+/// AnchorRequest for a Data Event on a Stream
+pub struct AnchorRequest {
+    /// The CID of the stream
+    pub id: Cid,
+    /// The CID of the Event to be anchored
+    pub prev: Cid,
+}
+
+/// Merkle tree node
+pub type MerkleNode = Vec<Cid>;
+
+/// A collection of Merkle tree nodes.
+#[derive(Default)]
+pub struct MerkleNodes {
+    nodes: IndexMap<Cid, MerkleNode>,
```

Comment:
Curious on the choice of an IndexMap? Should we be using an IndexMap anytime we have CID keys?

Reply:
This is to keep the merkel tree nodes in order.  A hash map will randomize the order.
Non-deterministic order makes testing hard.
Sorted order you will need to use later tree nodes to use earlier ones. Not nesaraly a problem just pulling more nodes into memory as we validate the time events.

Reply:
Ok so why not a simple BTreeMap in that case?

Reply:
Actually I think I misunderstood _order_. We need to keep the nodes in insertion order not just any stable order?

Reply:
Yeah, the order here is only used for debugging and expect tests. It makes it easier for us to know where there are issues because the remote nodes and local nodes remain in their respective groupings.